prometheus/promhttp/http.go

```diff
@@ -78,6 +92,56 @@ func Handler() http.Handler {
 	return HandlerFor(prometheus.DefaultGatherer, HandlerOpts{})
 }
 
+// InstrumentedHandler works in the same way as Handler but the returned HTTP
+// handler is instrumented with two metrics, both registered with the
+// DefaultRegistry: "promhttp_metric_handler_requests_total", a counter vec for
+// scrapes labeled with the HTTP status code, and
+// "promhttp_metric_handler_requests_in_flight", a gauge for the number of
+// simultaneous scrapes. Note that the Prometheus server already generates a
+// "scrape_duration_seconds" gauge, recording the duration of each scrape. The
+// metrics provided by InstrumentedHandler are useful to see how many scrapes
+// hit the monitored target (which could be from different Prometheus servers or
+// other scrapers), and how often they overlap (which would result in more than
+// one scrape in flight at the same time). Note that the in-flight gauge will
+// contain the scrape by which it is exposed, while the scrape counter will only
+// get incremented after the scrape is complete (as only then the status code is
+// known).
+//
+// InstrumentedHandler is meant to cover most of the simple cases. If you need
+// anything special, like other metrics or non-default HandlerOpts, or you are
+// not using the DefaultRegisterer, you have to use the various
+// InstrumentHandler… middlewares provided by this package to do the
+// instrumentation explicitly.
+//
+// InstrumentHandler panics if the registration of the metrics fails. If metrics
+// of the same name are already registered, the already registered ones are
+// tried, again panicking upon failure.
+func InstrumentedHandler() http.Handler {
```

Comment:
I don't think this will be a lot of use if users specifically have to turn it on, the benefit of these metrics is that they are there automatically.

Reply:
Users of the default registry pick either `Handler` or `InstrumentedHandler`. There is nothing more or less automatic picking the one over the other. I always hated to have an `UninstrumentedHandler` in the previous approach. The naming now is way more natural, readable, and understandable.

Reply:
My point is that users should not have to pick, this should be just one more metric users of the default registry get automatically like process_cpu_seconds_total. We should only have one Handler, and it should be instrumented - otherwise there's not much point having instrumentation as users won't use it.

Reply:
I don't follow the argument of "users won't use it". Users who can't get as far as seeing that there is `Handler` and `InstrumentedHandler` will hardly be able to make use of the provided handler metrics anyway.

However, I get the consistency point: The default registry already has process and go metrics, so the default handler should also have handler metrics. It's easy enough to create an uninstrumented handler with `HandlerFor` (and more involved use cases won't use the global registry anyway).

So yeah, let's just have one global `Handler`, which is instrumented. Unless @grobie has other thoughts, I'll change the commit later today.